File: b/Pc.php
<?php declare(strict_types = 1);

class Pc
{
	private Processor $ProcObj;
	private Disk $DiskObj;
	private Ram $RamObj;

	public function __construct($ProcObj, $DiskObj, $RamObj)
	{
		$this->ProcObj = $ProcObj;
		$this->DiskObj = $DiskObj;
		$this->RamObj =  $RamObj;
	}

	public function setProcessor(Processor $ProcObj): void { $this->ProcObj = $ProcObj; }
	public function setDisk(Disk $DiskObj): void { $this->DiskObj = $DiskObj; }
	public function setRam(Ram $RamObj): void { $this->RamObj = $RamObj; }

	public function getProcessor(): Processor { return $this->ProcObj; }
	public function getDisk(): Disk { return $this->DiskObj; }
	public function getRam(): Ram { return $this->RamObj; }

	public function printPc()
	{
		$this->ProcObj->printProcessor();
		$this->DiskObj->printDisk();
		$this->RamObj->printRam();
	}
}

?>

File: b/Pc.java
class Pc
{
	private Processor ProcObj;
	private Disk DiskObj;
	private Ram RamObj;

	// construct
	public Pc(Processor ProcObj, Disk DiskObj, Ram RamObj)
	{
		this.ProcObj = ProcObj;
		this.DiskObj = DiskObj;
		this.RamObj = RamObj;
	}

	// default construct
	public Pc(){}

	public void setProcessor(Processor ProcObj) { this.ProcObj = ProcObj; }
	public void setDisk(Disk DiskObj) { this.DiskObj = DiskObj; }
	public void setRam(Ram RamObj) { this.RamObj = RamObj; }

	public Processor getProcessor() { return this.ProcObj; }
	public Disk getDisk() { return this.DiskObj; }
	public Ram getRam() { return this.RamObj; }

	public void printPc()
	{
		ProcObj.printProcessor();
		DiskObj.printDisk();
		RamObj.printRam();
	}
}